internal/uvm/cpugroups.go

```diff
@@ -0,0 +1,158 @@
+package uvm
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/Microsoft/hcsshim/internal/hcs"
+	"github.com/Microsoft/hcsshim/internal/log"
+	hcsschema "github.com/Microsoft/hcsshim/internal/schema2"
+)
+
+const cpuGroupNullID = "00000000-0000-0000-0000-000000000000"
+
+var HV_STATUS_INVALID_CPU_GROUP_STATE = errors.New("The hypervisor could not perform the operation because the CPU group is entering or in an invalid state.")
+
+// ReleaseCPUGroup unsets the cpugroup from the VM and attemps to delete it
+func (uvm *UtilityVM) ReleaseCPUGroup(ctx context.Context) error {
+	groupID := uvm.cpuGroupID
+	if groupID == "" {
+		// not set, don't try to do anything
+		return nil
+	}
+	if err := uvm.unsetCPUGroup(ctx); err != nil {
+		return fmt.Errorf("failed to remove VM %s from cpugroup %s", uvm.ID(), groupID)
+	}
+
+	err := DeleteCPUGroup(ctx, groupID)
```

Comment:
Don't think we should be deleting the groups, since we expect them to be managed by the orchestrator.